### DISCUSSION TITLE
[BP#40] N'utilisez que les portions indispensables des librairies Javascript et frameworks CSS

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#83] [Bonnes pratiques] [answered]
Ces libs proposent souvent des CDNs ce qui permet de mutualiser de l'hébergement et du cache dans les navigateurs, je suis un peu criconspect du coup ? Ça ne me paraît pas aussi évident, quelqu'un aurait des sources?

A changer aussi: les libs citées sont viellissantes 😉 => plus récentes ou formulation plus intemporelle ?

**Comment by rlemaire** [comment]
C'est mieux d'héberger soit même les librairies. Si le CDN de la lib est down ou qu'ils changent le lien ou autre, on ne peut pas récupérer la dépendance et le site est cassé.
Avec des outils comme webpack on peut enlever automatiquement tout le code non utilisé, ce qui peut beaucoup réduire la taille des fichiers et donc la bande passante utilisée.

**Reply by devcrafting** [reply]
oki

**Comment by tbroyer** [comment]
Plutôt d'accord avec Raphaël. Les CDNs c'est bien mais

* c'est un domaine supplémentaire (BP#18)
* pour [limiter le tracking et prévenir certaines attaques](https://github.com/shivanigithub/http-cache-partitioning/#goals), les navigateurs s'orientent vers un cache partitionné qui vient mettre à mal cette idée de mutualisation du cache par les CDNs (https://chromestatus.com/features/5730772021411840, https://fetch.spec.whatwg.org/#http-cache-partitions)
* puisqu'on dépend d'un _third-party_, on introduit aussi des risques (sécurité), qui peuvent être limités ([SRI](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)) mais qu'il ne faut pas négliger.

On pourrait reformuler la dernière phrase pour la faire mieux correspondre à la réalité du terrain: “télécharger les sources et créer un build custom qui ne comprend que les composants nécessaires” aujourd'hui c'est utiliser un _bundler_ qui fait du _tree shaking_ / _dead code removal_. (https://bundlers.tooling.report/transformations/dead-code/, https://bundlers.tooling.report/transformations/dead-code-dynamic/)

Par contre, plutôt d'accord que mootools et YUI devraient être supprimées, mais jQuery reste largement utilisé aujourd'hui ; côté CSS Bootstrap a peut-être aujourd'hui d'autres concurrents que Foundation ? (je ne connais pas les autres) Tailwind par exemple ? Tachyons, Bulma ? (je ne m'y connais pas trop, je n'utilise pas ce genre d'outils)

**Reply by devcrafting** [reply]
je préfère ces propositions de formulation => à creuser :)

**Comment by ACTLEM** [comment, marked as answer]
Note équipe projet: Reformuler la dernière phrase pour mieux correspondre au terrain

Issue créée: https://github.com/cnumr/best-practices/issues/180
